### ISSUE TITLE
`Input.is_action_just_pressed()` may return true for multiple frames in a row when using joypad axis

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#83416]
### Godot version

4.2 beta1

### System information

Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1060 (NVIDIA; 30.0.15.1403) - Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 Threads)

### Issue description

```GDScript
func _process(delta):
	print(Input.is_action_just_pressed("ui_right"))
```
This may return true for single analog movement. It's a regression
(╯°□°)╯︵ ┻━┻

### Steps to reproduce

Use the code above with a controller (tested with Xbox One S controller). Make sure you have joypad axis assigned to an action.

### Minimal reproduction project

N/A

**Comment by Rubonnek** [comment]
I'm able to reproduce.

Holding the stick at the extreme of certain angle (roughly bottom-right) and moving it along the circumference will also make ` Input.is_action_just_pressed` return `true` for very minor changes in the stick angle. I wasn't expecting that.

Here's the code I used:

```gdscript
func _input(event):
	if (Input.is_action_just_pressed("ui_right")):
		print(event.as_text())

func _process(delta):
	print(Input.is_action_just_pressed("ui_right"))
```

Here's the relevant output:
```
Joypad Motion on Axis 0 (Left Stick X-Axis, Joystick 0 X-Axis) with Value 0.74
true
Joypad Motion on Axis 0 (Left Stick X-Axis, Joystick 0 X-Axis) with Value 0.73
Joypad Motion on Axis 1 (Left Stick Y-Axis, Joystick 0 Y-Axis) with Value 0.85
true
Joypad Motion on Axis 0 (Left Stick X-Axis, Joystick 0 X-Axis) with Value 0.68
Joypad Motion on Axis 1 (Left Stick Y-Axis, Joystick 0 Y-Axis) with Value 0.91
true
Joypad Motion on Axis 0 (Left Stick X-Axis, Joystick 0 X-Axis) with Value 0.66
Joypad Motion on Axis 1 (Left Stick Y-Axis, Joystick 0 Y-Axis) with Value 0.93
true
Joypad Motion on Axis 0 (Left Stick X-Axis, Joystick 0 X-Axis) with Value 0.63
Joypad Motion on Axis 1 (Left Stick Y-Axis, Joystick 0 Y-Axis) with Value 0.94
true
Joypad Motion on Axis 0 (Left Stick X-Axis, Joystick 0 X-Axis) with Value 0.60
Joypad Motion on Axis 1 (Left Stick Y-Axis, Joystick 0 Y-Axis) with Value 0.95
```

**Comment by mieldepoche** [comment]
duplicate of https://github.com/godotengine/godot/issues/81975 ?

**Comment by Rubonnek** [comment]
> duplicate of #81975 ?

My comment above was definitely related to that issue. I've marked my comment as off-topic.

However, there's a chance this issue is specific to Windows, unless it is fixed by #82056.

**Comment by KoBeWi** [comment]
That PR fixes it, so closing as duplicate.
